src/coreComponents/mesh/CellElementRegion.cpp

```diff
@@ -35,16 +73,163 @@ CellElementRegion::CellElementRegion( string const & name, Group * const parent
 CellElementRegion::~CellElementRegion()
 {}
 
-void CellElementRegion::generateMesh( Group const & cellBlocks )
+
+void CellElementRegion::registerSubRegion( CellBlockABC const & cellBlock )
 {
   Group & elementSubRegions = this->getGroup( viewKeyStruct::elementSubRegions() );
+  // For now, subRegion name must be the same as the cellBlock (so we can match them and reference them in errors).
+  CellElementSubRegion & subRegion =
+    elementSubRegions.registerGroup< CellElementSubRegion >( cellBlock.getName() );
+
+  subRegion.copyFromCellBlock( cellBlock );
+}
+
+string CellElementRegion::getCellBlockAttributeValue( string_view cellBlockName )
+{
+  return string( stringutilities::removeStringAndFollowingContent( cellBlockName, "_" ) );
```

Comment:
I though I could use strings to store the region attribute for these reasons:
- we do not really need integers here, there are no numerical operation at all,
- to avoid unnecessary parsing / formatting operations,
- to support text region names in the future.